tables/file.py

```diff
@@ -2849,7 +2849,7 @@ def _update_node_locations(self, oldpath, newpath):
 
         # Update alive and dead descendents.
         for cache in [self._node_manager.cache, self._node_manager.registry]:
-            for nodepath in cache:
+            for nodepath in list(cache):
```

Comment:
Does `iter(cache)` or `cache.keys()` work here instead? Or does it need to be a list?

Reply:
It needs to be a copy. List, tuple, etc. Iter or keys won't help.